### PR TITLE
Use zipped xcframework in nuget package

### PR DESCRIPTION
**Pull request by skottmckay**
### Description

The xcframework now uses symlinks to have the correct structure according to Apple requirements. Symlinks are not supported by nuget on Windows.

In order to work around that we can store a zip of the xcframeworks in the nuget package.

### Motivation and Context

Fix nuget packaging build break

### CODE REVIEW

File: tools/nuget/generate_nuspec_for_native_nuget.py [hunk 1]
```diff
@@ -105,8 +105,10 @@ def generate_file_list_for_ep(nuget_artifacts_dir, ep, files_list, include_pdbs,
                 if child_file.suffix in [".aar"]:
                     files_list.append('<file src="' + str(child_file) + '" target="runtimes/android/native"/>')
 
-        if child.name == "onnxruntime-ios-xcframework":
-            files_list.append('<file src="' + str(child) + "\\**" '" target="runtimes/ios/native"/>')  # noqa: ISC001
+        if child.name == "onnxruntime-ios":
```
Comment (edgchen1) [inline tools/nuget/generate_nuspec_for_native_nuget.py:108]: where does this "onnxruntime-ios" directory name come from?

File: tools/ci_build/github/windows/extract_nuget_files.ps1 [hunk 2]
```diff
@@ -34,6 +35,23 @@ Foreach-Object {
...
+  $target_dir = "$nuget_artifacts_dir\onnxruntime-ios-xcframework"
+  # remove version info from filename and use required filename format
+  $target_file = "$target_dir\onnxruntime.xcframework.zip"
+  New-Item -Path $target_dir -ItemType directory
+
+  Write-Output "Copy-Item $($xcframework.FullName) $target_file"
+  Copy-Item $xcframework.FullName $target_file
+}
+elseif ($xcframeworks.Count -gt 1) {
+  Write-Error "Expected at most one onnxruntime_ios_xcframework*.zip file but got: [$xcframeworks]"
```
Comment (edgchen1) [inline tools/ci_build/github/windows/extract_nuget_files.ps1:51]: nit: to match the filter
```suggestion
  Write-Error "Expected at most one onnxruntime_ios_xcframework.*.zip file but got: [$xcframeworks]"
```

**Review by edgchen1 — APPROVED** [review]

File: tools/nuget/generate_nuspec_for_native_nuget.py (hunk 1 above)
Comment (skottmckay) [reply tools/nuget/generate_nuspec_for_native_nuget.py:108]: there's a loop to trim the trailing '-xxx' off each directory name at the end of the extract script. that in general handles version numbers for most of the packages. existing code was trimming onnxruntime-android-aar to onnxruntime-android, so it's somewhat consistent with that to also let it trim onnxruntime-ios-xcframework to onnxruntime-ios. definitely cryptic though

there's is too much implicit stuff between the extraction script logic and the required structure here. would probably be simpler if the extract was done in python here.

Comment (edgchen1) [reply tools/nuget/generate_nuspec_for_native_nuget.py:108]: ah ok, thanks for the explanation.

**Review by mszhanyi — APPROVED** [review]

**Pull request merged**